packages/wallet/backend/src/account/service.ts

```diff
@@ -70,51 +61,25 @@ export class AccountService implements IAccountService {
 
     // issue virtual account to wallet
     const user = await User.query().findById(args.userId)
-    if (!user) {
+    if (!user || !user.gateHubUserId) {
       throw new NotFound()
     }
 
-    const result = await this.rapydClient.issueVirtualAccount({
-      country: user.country ?? '',
-      currency: asset.code,
-      ewallet: user.rapydWalletId ?? ''
-    })
-
-    if (result.status?.status !== 'SUCCESS') {
-      throw new Error(
-        `Unable to issue virtal account to ewallet: ${result.status?.message}`
-      )
-    }
-    // save virtual bank account number to database
-    const virtualAccount = result.data
+    const result = await this.gateHubClient.createWallet(
+      user.gateHubUserId,
+      args.name
+    )
```

Comment:
should we use one wallet for all the assets?

Reply:
I thought that a wallet for each currency may be easier to manage

Reply:
> should we use one wallet for all the assets?

CC @beniaminmunteanu @sanducb

Reply:
IMO having one wallet for each currency indeed seems easier to manage (as suggested by @dragosp1011). Also, looking into the GateHub [docs](https://docs.gatehub.net/api-documentation/c3OPAp5dM191CDAdwyYS/api-reference/api-reference/wallets):
```
Wallets are accounts that can hold one or more assets.
```
I would say that since for GateHub wallets are considered accounts, then we could use the same account abstraction on our side and respect our own rules i.e. having one currency per account.